Tools/TestWebKitAPI/Tests/WebCore/MIMESniffer.cpp

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+
+#include <WebCore/MIMESniffer.h>
+
+using namespace WebCore;
+
+namespace TestWebKitAPI {
+
+TEST(MIMESniffer, MIMETypes)
+{
+    static std::span<const uint8_t> oggData = { { 'O', 'g', 'g', 'S', 0, 'm', 'e', 'e', 'e', 'h', '.' } };
```

Comment:
I am really not convinced this is safe. I believe your span is dangling on the next line since it is constructed from an initializer list which is gone on the next line.

Reply:
it is not an initialiser list, this is only added in C++26 (https://en.cppreference.com/w/cpp/container/span/span)
I can re-work it if that makes it clear.

Reply:
Ok, it's using the constructor from an array. I think my point stands that it is unsafe / UB:
https://quuxplusone.github.io/blog/2021/10/03/p2447-span-from-initializer-list/

```
std::span<const int> sp = {{1, 2, 3}};  // OK
for (int i : sp) std::cout << i;  // UB, possible segfault
```